indico/modules/events/registration/models/invitations.py

```diff
@@ -80,6 +80,12 @@ class RegistrationInvitation(db.Model):
         nullable=False,
         default=False
     )
+    #: Whether registration email should match invitation email
+    allow_different_email = db.Column(
```

Comment:
Just wondering, didn't you at some point also consider locking the name fields?
I'd be open to this being an array of locked field names, and then using a checkbox list in the form where you tick firstname/lastname/email and whatever you ticked there gets locked.